### PR TITLE
Fix Scatterplot with specific combination of variables

### LINKED ISSUE #192: PopEx: Variable available for grouping, but can't be plotted
This variable
image

... doesn't work in Scatterplot (probably because values are "Y" and `NA`. I probably just need to convert `NA` --> "NA"):
image

... but Works in boxplot & line plot:
image

### PR DESCRIPTION
**Pull request by AARON-CLARK**
This PR accomplishes a few tasks:
* fixes #192
* re-orgs plot type order
* polishes code to be more succinct
* adds variable to strip label when faceting plot by another variable's values

### CODE REVIEW

**Comment by AARON-CLARK** [comment]
@Jeff-Thompson12, I think this solution works. Could you take a look?

File: R/mod_popExp_fct_scatterplot.R [hunk 1]
```diff
@@ -132,7 +132,8 @@ app_scatterplot <- function(data, yvar, xvar, week_x, value_x, week_y, value_y,
         dplyr::select(USUBJID, PARAMCD, value_y, one_of(color, separate)) %>%
         tidyr::pivot_wider(names_from = PARAMCD, values_from = value_y) %>%
         tidyr::unnest(yvar) %>% # if their are more than 1 AVAL per Patient, per Visit
-        select_if(~!all(is.na(.))) # remove NA cols
+        dplyr::mutate(across(where(function(x) all(is.na(x))), ~ "NA" )) #%>% # Convert NA cols
+        # select_if(~!all(is.na(.))) # remove NA cols
```
Comment (Jeff-Thompson12) [inline R/mod_popExp_fct_scatterplot.R:136]: What is the purpose of `select_if(~!all(is.na(.)))`? The solution I see here would be achieved by just commenting that line out completely. I'm assuming it is meant to handle a situation that can arise from `pivot_wider()`. Maybe the mutate should only be applied to `color` and `separate`.

**Review by Jeff-Thompson12 — COMMENTED** [review]
@AARON-CLARK The thing that is tripping me up is that I don't know what situations the `select_if()` was meant to fix/prevent in the first place. Do you recall why that line of code was originally needed?

File: R/mod_popExp_fct_scatterplot.R (hunk 1 above)
Comment (Jeff-Thompson12) [reply R/mod_popExp_fct_scatterplot.R:136]: I see that it is not equivalent to commenting out the last line. Specifically the color grouping will not work if the field is completely missing (i.e. `all(is.na(.))`)

**Comment by Jeff-Thompson12** [comment]
Levels should be used in the labels if available.
image

**Comment by AARON-CLARK** [comment]
Oh my mistake. I already fixed that problem locally. Let me make sure I've pushed all my commits.

**Comment by AARON-CLARK** [comment]
@Jeff-Thompson12, should be fixed now. Good catch.

image

**Review by Jeff-Thompson12 — APPROVED** [review]
This seems pretty good to me.

**Pull request merged**